api/controllers/posts.js

```diff
@@ -22,11 +30,29 @@ const updateLikes = async (req, res) => {
 };
 
 const createPost = async (req, res) => {
-  const post = new Post(req.body);
-  post.save();
+  const user_id = req.user_id; 
+  const user = await User.findById(user_id);
+  console.log(user)
+
+
+  if (!user) {
+    return res.status(404).json({ message: "User not found" });
+  }
+
+  const email = user.email;
+  console.log(email)
+
+  const post = new Post({
+    ...req.body,
+    user: user._id,
+    email: user.email  
+  });
+  await post.save();
+
+  console.log("Post created:", post); 
 
   const newToken = generateToken(req.user_id);
-  res.status(201).json({ message: "Post created", token: newToken });
+  res.status(201).json({ message: "Post created", email: user.email, token: newToken });
```

Comment:
email gets declared above (line 42), can use that.
However, we're giving users the ability to change their email, so this would potentially store obsolete information.
It's probably best to only store the user's ID as that never changes.